pkg/registry/core/service/storage/rest_test.go

```diff
@@ -279,6 +287,41 @@ func TestServiceRegistryCreate(t *testing.T) {
 	}
 }
 
+func TestServiceRegistryCreateDryRun(t *testing.T) {
+	storage, registry, server := NewTestREST(t, nil)
+	defer server.Terminate(t)
+
+	svc := &api.Service{
+		ObjectMeta: metav1.ObjectMeta{Name: "foo"},
+		Spec: api.ServiceSpec{
+			Selector:        map[string]string{"bar": "baz"},
+			SessionAffinity: api.ServiceAffinityNone,
+			Type:            api.ServiceTypeNodePort,
+			Ports: []api.ServicePort{{
+				NodePort:   30010,
+				Port:       6502,
+				Protocol:   api.ProtocolTCP,
+				TargetPort: intstr.FromInt(6502),
+			}},
+		},
+	}
+	ctx := genericapirequest.NewDefaultContext()
+	_, err := storage.Create(ctx, svc, rest.ValidateAllObjectFunc, &metav1.CreateOptions{DryRun: []string{metav1.DryRunAll}})
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	if storage.serviceNodePorts.Has(30010) {
```

Comment:
Don't services also have a cluster IP? We should test that, also.

This testing mechanism is good but I wonder if it'd be better to add a mechanism to the fake storage to verify that nothing changed at all, either by ensuring no calls were made or by comparing pre/post state.

Reply:
I don't think we need to test the underlying storage beyond making sure that the dry run flag gets passed along to it, since that behavior is tested elsewhere.